opensearch-api/lib/opensearch/api/utils.rb

```diff
@@ -40,7 +40,7 @@ module Utils
       # @api private
       def __escape(string)
         return string if string == '*'
-        defined?(EscapeUtils) ? EscapeUtils.escape_url(string.to_s) : CGI.escape(string.to_s)
+        defined?(EscapeUtils) ? EscapeUtils.escape_uri(string.to_s) : CGI.escape(string.to_s)
```

Comment:
Curious why we had to change to use URI ?

Reply:
Nvm I just saw the comment above.

Reply:
I removed it in https://github.com/opensearch-project/opensearch-ruby/pull/74